src/particles/deposition/ExplicitDeposition.cpp

```diff
@@ -96,75 +107,46 @@ ExplicitDeposition (PlasmaParticleContainer& plasma, Fields& fields, const Multi
-                    for (int iy=0; iy<=depos_order; iy++){
-                        for (int ix=0; ix<=depos_order; ix++){
-                            // Its important that Aabssqp is first fully gathered and not used
-                            // directly per cell like AabssqDxp and AabssqDyp
-                            const amrex::Real x00y00 = abssq(
-                                laser_arr(i_cell+ix, j_cell+iy, 0),
-                                laser_arr(i_cell+ix, j_cell+iy, 1) );
-                            Aabssqp += sx_cell[ix]*sy_cell[iy]*x00y00;
-                        }
-                    }
+                    // Its important that Aabssqp is first fully gathered and not used
+                    // directly per cell like AabssqDxp and AabssqDyp
+                    doLaserGatherShapeN<depos_order>(xp, yp, Aabssqp, laser_arr,
+                                                     dx_inv, dy_inv, x_pos_offset, y_pos_offset);
                 }
 
                 // calculate gamma/psi for plasma particles
                 const amrex::Real gamma_psi = 0.5_rt * (
-                    (1._rt + 0.5_rt * Aabssqp) / (psi * psi) // TODO: fix units
+                    (1._rt + 0.5_rt * Aabssqp) * psi_inv * psi_inv // TODO: fix units
                     + vx * vx
                     + vy * vy
                     + 1._rt
                 );
 
-                for (int iy=0; iy <= depos_order+2; ++iy) {
-                    // normal shape factor
-                    amrex::Real shape_y = 0._rt;
-                    // derivative shape factor
-                    amrex::Real shape_dy = 0._rt;
-                    if (iy != 0 && iy != depos_order + 2) {
-                        shape_y = sy_cell[iy-1] * charge_density_mu0;
-                    }
-                    if (iy < depos_order + 1) {
-                        shape_dy = sy_cell[iy];
-                    }
-                    if (iy > 1) {
-                        shape_dy -= sy_cell[iy-2];
-                    }
-                    shape_dy *= dy_inv * 0.5_rt * clight * charge_density_mu0;
-
-                    for (int ix=0; ix <= depos_order+2; ++ix) {
-                        amrex::Real shape_x = 0._rt;
-                        amrex::Real shape_dx = 0._rt;
-                        if (ix != 0 && ix != depos_order + 2) {
-                            shape_x = sx_cell[ix-1];
-                        }
-                        if (ix < depos_order + 1) {
-                            shape_dx = sx_cell[ix];
-                        }
-                        if (ix > 1) {
-                            shape_dx -= sx_cell[ix-2];
-                        }
-                        shape_dx *= dx_inv * 0.5_rt * clight;
-
-                        if ((ix==0 || ix==depos_order + 2) && (iy==0 || iy==depos_order + 2)) {
-                            // corners have a shape factor of zero
-                            continue;
+#ifdef AMREX_USE_GPU
+#pragma unroll
```

Comment:
Disregard since we don’t do this for other pragmas.
```
alex@hipace$ git grep "pragma"
src/fields/Fields.cpp:#pragma omp parallel
src/fields/Fields.cpp:#pragma omp parallel if (amrex::Gpu::notInLaunchRegion())
src/fields/Fields.cpp:#pragma omp parallel if (amrex::Gpu::notInLaunchRegion())
src/laser/MultiLaser.cpp:#pragma omp parallel if (amrex::Gpu::notInLaunchRegion())
src/mg_solver/HpMultiGrid.H:#pragma omp parallel for simd
src/mg_solver/HpMultiGrid.H:#pragma omp parallel for collapse(2)
src/mg_solver/HpMultiGrid.H:#pragma omp simd
src/mg_solver/HpMultiGrid.H:#pragma omp parallel for collapse(3)
src/mg_solver/HpMultiGrid.H:#pragma omp simd
src/particles/deposition/ExplicitDeposition.cpp:#pragma unroll
src/particles/deposition/ExplicitDeposition.cpp:#pragma unroll
src/particles/deposition/PlasmaDepositCurrent.cpp:#pragma omp parallel if (amrex::Gpu::notInLaunchRegion())
src/particles/deposition/PlasmaDepositCurrent.cpp:#pragma omp for
src/particles/pusher/PlasmaParticleAdvance.cpp:#pragma omp parallel for if (amrex::Gpu::notInLaunchRegion())
src/salame/Salame.cpp:#pragma omp parallel for if (amrex::Gpu::notInLaunchRegion())
```